h2o-core/src/main/java/water/api/ModelBuilderHandler.java

```diff
@@ -38,11 +39,23 @@ public S handle(int version, Route route, Properties parms, String postBody) thr
 
     // User specified key, or make a default?
     String model_id = parms.getProperty("model_id");
+    String warningStr = null;
```

Comment:
True.  Will check if warningStr is null or not instead of having another variable.  New constructor is needed per our discussion.